test/fixtures/spell-check/success0.md

```diff
@@ -2,27 +2,39 @@ This file should not generate any spell-check errors.
 
 ## Node.js
 
-[Node.js](https://foo.com) - Node.js
+[Node.js](https://nodejs.com) - Node.js
 
 ## Stack Overflow
 
-[Stack Overflow](https://foo.com) - Stack Overflow
+[Stack Overflow](https://stackoverflow.com) - Stack Overflow
 
 ## JavaScript
 
-[JavaScript](https://foo.com) - JavaScript
+[JavaScript](https://javascript.com) - JavaScript
 
 ## macOS
 
-[macOS](https://foo.com) - macOS
+[macOS](https://macos.com) - macOS
 
 ## YouTube
 
-[YouTube](https://foo.com) - YouTube
+[YouTube](https://youtube.com) - YouTube
 
 ## GitHub
 
-[GitHub](https://foo.com) - GitHub
+[GitHub](https://github.com) - GitHub
+
+## GitHub as a standalone URL
+
+github.com
+
+## GitHub as a standalone URL with path
+
+https://github.com/sindresorhus/awesome
+
+## GitHub as a URL for a link-text
+
+[https://github.com/sindresorhus/awesome](https://github.com/sindresorhus/awesome)
```

Comment:
@transitive-bullshit here are [the examples you'd asked about](https://github.com/sindresorhus/awesome-lint/issues/71)